### DISCUSSION TITLE
Add parent directory as a tag

### DISCUSSION THREAD

**Discussion by petazonk** [deckerst/aves#127] [Q&A] [answered]
I have my photos categorised under some directories like Camping, Holidays, Weddings and so on. If each of these parent directories was added as a tag then I could easily select the Camping tag to see all camping photos. Or, is there another way to do this as I couldn't find out how to navigate directories?

**Comment by deckerst** [comment, marked as answer]
There is currently no way to open a subtree or only show media from a subtree. An album only reflects the direct folder containing media items.

I don't think automatic tagging with parent directories is the best solution (if I correctly understand the suggestion). It would be surprising or ambiguous for many people. Also, if you have a folder like "/sdcard/0/Pictures/Holidays/Asia/Japan trip", should the app add a tag for "Asia" but also "Holidays", "Pictures"... "0", "sdcard"?

I'm thinking of 3 ways this could be handled:
1) __album groups__: at the moment, the Albums page shows a flat list of folder albums. It could be modified to allow grouping albums (like you group apps on your phone).
    - Pro: it's flexible so that you could group folders that are not even in the same subtree, or only some folders in the subtree.
    - Con: cluttered UI/gestures with actions to group/ungroup, view group or view its components, etc. Also, it wouldn't be dynamic, so if you add a new media folder in your "Camping" directory, you would need to manually add it to the album group.
2) __virtual albums__: along with the current folder albums, the app would allow creating albums based on other criteria. For example, creating a virtual album gathering photos with a specific tag taken in a specific place. One of these criteria could be a directory path, like the "Camping" directory.
    - Pro: dynamic, as you only need to set up the virtual album once, and any future media matching the criteria would be automatically part of the virtual album.
    -  Con: some users may get confused by the concepts of "(folder) albums" / "virtual albums".
3) __XMP tags__: if you already use tags to manage your collection, it's not much work to just tag all your camping media with "camping", and the app already knows how to detect and display these tags.
    - Pro: no additional features or UI needed.
    - Con: it requires modifying the media files to include the tags.

I'm not a fan of album groups, but I will likely implement the virtual albums at some point.

**Comment by petazonk** [comment]
Yes I agree with everything you have said and so I think I do need to just update my tags. What is the preferred tag that Aves would recognise for say Camping, Weddings and so on?

**Reply by deckerst** [reply]
The specific words? It could be anything. It could even be an emoji if you want. The tags themselves are just used as they are, they are not semantically interpreted.

If you're unsure, I suggest you just try to tag a few disposable pictures and see how tags appear in Aves.

By the way, I'm actively working on tag *editing* right now. Hopefully this will be ready for the next release, at least to tag the most common image formats.

**Reply by petazonk** [reply]
But there are multiple tags from EXIF, IPTC, XMP and so on. Also hierarchical like People|Joe Blogs and so on. I can research it, but was just wondering if there is a more popular one. I guess I could just use DigiKam to do it and see how I go.

**Reply by deckerst** [reply]
Usually, applications that can tag (like Picasa and DigiKam) write tags in XMP and IPTC. Tags (in that sense of "labels") are not stored in EXIF. XMP supports flat tags as well as hierarchical tags, but IPTC only supports flat tags.

Aves extracts tags from the following (by precedence):
- in XMP, the `dc:subject` properties,
- in IPTC, the `keywords` field.

Aves does not currently support the hierarchical tags. It's something I will consider if there is demand. AFAIK hierarchical tags are properly stored in XMP's `lr:hierarchicalSubject` or `mwg-kw:Keywords` properties, but I suppose they may be stored in other fields too, with the `|` character like in your example (the MWG recommends to flatten tags stored in `dc:subject` though).

**Reply by deckerst** [reply]
Tags are editable in Aves v1.5.7.

It's limited to a few formats, cf [wiki](https://github.com/deckerst/aves/wiki/Editing-Support).

**Reply by petazonk** [reply]
I was just wondering if you have had any more thoughts on indexing Picasa people tags? eg MWG I think. I have done most of my face tagging with Picasa and so I have tens of thousands of face tags that I can't access with Aves :(

**Reply by deckerst** [reply]
It's tracked by #69. Unless it's closed or tagged `P6` it's something I'd like to do. I'll get to it when I have the time and energy to do so. No timelines, no promises!
